src/network/tc/tc-util.c

```diff
@@ -92,3 +93,25 @@ int tc_fill_ratespec_and_table(struct tc_ratespec *rate, uint32_t *rtab, uint32_
         rate->linklayer = TC_LINKLAYER_ETHERNET;
         return 0;
 }
+
+int tc_parse_qdisc_handle(const char *s, uint32_t *h) {
+        uint32_t m;
+        char *p;
+
+        m = TC_H_UNSPEC;
+        if (streq(s, "none")) {
+                *h = m;
+                return 0;
+        }
+
+        m = strtoul(s, &p, 16);
```

Comment:
strtoul returns "unsigned long". Which is not the same as "uint32_t". Please be more careful with the types. 

Only safe way to call strtoul() is by setting errno = 0 first, and checking if it is set afterwards.

Please introduce safe_atou32_full() similar in style to safe_atou16_full() and then use that for parsing this. strtoul() is so hard to use right, hence we really should always use our wrappers.

Reply:
Sorry for late reply . I was travelling. It's a 32 bit number . But we can use safe_atou16_full() then assign this back .